sotodlib/io/load_smurf.py

```diff
@@ -1468,12 +1616,25 @@ def from_time(cls, time, archive, show_pb=False):
         """
         time = archive._make_datetime(time)
         session = archive.Session()
-        session_start,  = session.query(Frames.time).filter(
+        q  = session.query(Frames).filter(
                 Frames.type_name == 'Observation',
                 Frames.time <= time
-            ).order_by(Frames.time.desc()).first()
+            ).order_by(Frames.time.desc())
 
-        status_frames = session.query(Frames).filter(
+        if stream_id is not None:
+            q = q.join(Files).filter(Files.stream_id==stream_id)
+        else:
+            logger.warning(f"No stream_id given for finding status, not checking for one")
```

Comment:
You can use the `_stream_ids_in_range` function to check if there are multiple stream-id options at a given time. I think if no stream_id is passed and there are multiple options this should raise an error.